agent/hcp/client/client.go

```diff
@@ -329,3 +332,32 @@ func (t *TelemetryConfig) DefaultLabels(cfg config.CloudConfig) map[string]strin
 
 	return labels
 }
+
+// FilterRegex returns a valid regex used to filter metrics.
```

Comment:
This was just moved out of the telemetry package from `filter.go`. 

It can be used easily as a method on the `TelemetryConfig`.

For example, in `deps.go`:
https://github.com/hashicorp/consul/pull/18168/files#diff-d652bfc7ebdcf8edacd32ad13fd9059b666981a2a76ef9bc5d985be7edac5601R86-R89


This is specific for our use case so it's a nice cleanup. Someone using filters can provide a regex directly with their own provider.